### ISSUE TITLE
No LuaRocks module found for densecap-master/densecap/DenseCapModel

### ISSUE THREAD

**Issue by zs72** [jcjohnson/densecap#88]
Hi
I run your code. I facing this error.

$ th run_model.lua -input_image imgs/elephant.jpg
/home/mrrl/torch/install/bin/luajit: /home/mrrl/torch/install/share/lua/5.1/trepl/init.lua:389: module 'densecap-master/densecap/DenseCapModel' not found:No LuaRocks module found for densecap-master/densecap/DenseCapModel
no field package.preload['densecap-master/densecap/DenseCapModel']
no file '/home/mrrl/.luarocks/share/lua/5.1/densecap-master/densecap/DenseCapModel.lua'
no file '/home/mrrl/.luarocks/share/lua/5.1/densecap-master/densecap/DenseCapModel/init.lua'
no file '/home/mrrl/torch/install/share/lua/5.1/densecap-master/densecap/DenseCapModel.lua'
no file '/home/mrrl/torch/install/share/lua/5.1/densecap-master/densecap/DenseCapModel/init.lua'
no file './densecap-master/densecap/DenseCapModel.lua'
no file '/home/mrrl/torch/install/share/luajit-2.1.0-beta1/densecap-master/densecap/DenseCapModel.lua'
no file '/usr/local/share/lua/5.1/densecap-master/densecap/DenseCapModel.lua'
no file '/usr/local/share/lua/5.1/densecap-master/densecap/DenseCapModel/init.lua'
no file '/home/mrrl/.luarocks/lib/lua/5.1/densecap-master/densecap/DenseCapModel.so'
no file '/home/mrrl/torch/install/lib/lua/5.1/densecap-master/densecap/DenseCapModel.so'
no file '/home/mrrl/torch/install/lib/densecap-master/densecap/DenseCapModel.so'
no file './densecap-master/densecap/DenseCapModel.so'
no file '/usr/local/lib/lua/5.1/densecap-master/densecap/DenseCapModel.so'
no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
[C]: in function 'error'
/home/mrrl/torch/install/share/lua/5.1/trepl/init.lua:389: in function 'require'
run_model.lua:5: in main chunk
[C]: in function 'dofile'
...mrrl/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
[C]: at 0x00406980

can you help me?

**Comment by lahorite** [comment]
I am seeing similar error -
~# th densecap/run_model.lua -input_image densecap/imgs/elephant.jpg
/root/torch/install/bin/luajit: /root/torch/install/share/lua/5.1/trepl/init.lua:389: module 'densecap.DenseCapModel' not found:No LuaRocks module found for densecap.DenseCapModel
        no field package.preload['densecap.DenseCapModel']
        no file '/root/.luarocks/share/lua/5.1/densecap/DenseCapModel.lua'
        no file '/root/.luarocks/share/lua/5.1/densecap/DenseCapModel/init.lua'
        no file '/root/torch/install/share/lua/5.1/densecap/DenseCapModel.lua'
        no file '/root/torch/install/share/lua/5.1/densecap/DenseCapModel/init.lua'
        no file './densecap/DenseCapModel.lua'
        no file '/root/torch/install/share/luajit-2.1.0-beta1/densecap/DenseCapModel.lua'
        no file '/usr/local/share/lua/5.1/densecap/DenseCapModel.lua'
        no file '/usr/local/share/lua/5.1/densecap/DenseCapModel/init.lua'
        no file '/root/.luarocks/lib/lua/5.1/densecap/DenseCapModel.so'
        no file '/root/torch/install/lib/lua/5.1/densecap/DenseCapModel.so'
        no file '/root/torch/install/lib/densecap/DenseCapModel.so'
        no file './densecap/DenseCapModel.so'
        no file '/usr/local/lib/lua/5.1/densecap/DenseCapModel.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
        no file '/root/.luarocks/lib/lua/5.1/densecap.so'
        no file '/root/torch/install/lib/lua/5.1/densecap.so'
        no file '/root/torch/install/lib/densecap.so'
        no file './densecap.so'
        no file '/usr/local/lib/lua/5.1/densecap.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
stack traceback:
        [C]: in function 'error'
        /root/torch/install/share/lua/5.1/trepl/init.lua:389: in function 'require'
        densecap/run_model.lua:5: in main chunk
        [C]: in function 'dofile'
        /root/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:150: in main chunk
        [C]: at 0x5559ff7e3570

**Comment by jainal09** [comment]
Hey,
Check this docker image I created for running dense cap in gpu!
https://hub.docker.com/r/jainal09/densecap_gpu

For Cpu version.
Use this docker file.
https://github.com/runwayml/alpha_models/blob/master/densecap/Dockerfile

**Comment by ZanderBrumbaugh** [comment]
I'm currently having this exact error as well, it's the last thing preventing me from having a fully working version of DenseCap in Google Colab. It seems that maybe the require function isn't working as expected? It's looking for files that definitely aren't there.

**Comment by jainal09** [comment]
@ZanderBrumbaugh colab uses different ubuntu version this project requires ubuntu 14.0 to compile. Hence try the above docker images.

**Comment by ZanderBrumbaugh** [comment]
Interesting. I have DenseCap fully working on my Ubunutu VM which is version 20.04. I'll look into it tomorrow and ask any further questions if they come up. Thanks.

**Comment by ZanderBrumbaugh** [comment]
Yeah, I wasn't able to get Docker or Docker-py working in Google Colab either. That's pretty much the whole critical element of my research, making this tech really accessible. A notebook is the best way to do that and I've gotten the whole program working except for that one require error which is looking for files that aren't there. As I mentioned, it works in 20.04, so I feel there's a solution, even if it involves editing some files.

**Comment by cspearl** [comment]
> Interesting. I have DenseCap fully working on my Ubunutu VM which is version 20.04. I'll look into it tomorrow and ask any further questions if they come up. Thanks.

I am trying to run densecap on my ubuntu 20.04 but I am not able to install the dependencies properly and thus can't run the run_model.lua file. I dont have any gpu and trying to run it on cpu only - on custom images.
Also did you find any solution on running it in colab? Let me know. Thanks!

**Comment by malinfei** [comment]
这是来自名扬天下的QQ邮箱的自动回复！您好，您的邮件已经发送至名扬天下的邮箱，我会尽快查收。
